lib/puppet-syntax.rb

```diff
@@ -7,8 +7,9 @@ module PuppetSyntax
   @exclude_paths = []
   @future_parser = false
   @hieradata_paths = ["**/data/**/*.yaml", "hieradata/**/*.yaml", "hiera*.yaml"]
+  @fail_on_warnings = true
 
   class << self
-    attr_accessor :exclude_paths, :future_parser, :hieradata_paths
```

Comment:
Could `fail_on_warnings` be renamed to `fail_on_deprecation_notices`? The reason being that the logic in `lib/puppet-syntax/manifests.rb` distinguishes between deprecation notices and other errors and not warnings specifically.